### DISCUSSION TITLE
升级了umi4之后, 访问.wasm文件失败

### DISCUSSION THREAD

**Discussion by deepdeepstudy** [umijs/umi#8541] [Q&A]
引入rocket文件夹中的wasm文件
![截屏2022-07-14 下午8 24 20](https://user-images.githubusercontent.com/26989027/178981581-cace6a4f-04b0-4b4c-b133-1fa0463634cf.png)

import('../../../public/rocket/index.js')
  .then((module) => {
    const rocket: IRocket = {};
    Object.assign(rocket, module);
    const ver = rocket.ver();
    console.log('rocket verision:', ver);
    window.maliang.rocket = rocket;
  })
  .catch(console.error);

提示Can't import the named export '__wbindgen_start' (imported as 'wasm') from default-exporting module (only default export is available)

**Comment by deepdeepstudy** [comment]
有没有走通过引入wasm文件的同学，支个招，另外chainWebpack参数为空了，怎么写配置呢？

**Comment by fz6m** [comment]
给个复现 demo 看看。

**Reply by deepdeepstudy** [reply]
复现demo 
https://github.com/deepdeepstudy/test-umi4-wasm

**Reply by deepdeepstudy** [reply]
Module not found: Error: No parser registered for webassembly/async

**Comment by deepdeepstudy** [comment]
your e-mail is recieved，have a nice day!

**Comment by fz6m** [comment]
基于你的复现调整了一下。See https://github.com/MoeYc/umi4-wasm-example

主要调整：

 - `chainWebpack` 配置 wasm 规则，此处使用 webpack5 原生 wasm 加载能力，有 sync 和 async 两种，参考 [这里](https://toutiao.io/posts/qlynesm/preview) 。

 - 我这里配置了 async 的 wasm 规则之后，直接在项目里导入就可以了，webpack 会帮你 async 加载，所以不能用 `babel-plugin-dynamic-import-node` （构建产物无法运行，这个库比较老，一些场景处理不到），如果你有打包为单文件的诉求，应该尝试配成 sync 的规则，并且用 `import()` 尝试？我没试过，自己研究下吧。

 - 基于上一条，所以 wasm 不需要放到 public 里，直接作为项目的一部分，放到 src 里就行，webpack 打包会把 `.wasm` 文件作为静态资源带出来。

参考调整 commit ：https://github.com/MoeYc/umi4-wasm-example/commit/062565ab8173b4a70be73dcbd3272feda3d20cfe

**Reply by deepdeepstudy** [reply]
感谢大佬的灵丹妙药！很棒！

**Reply by Linefate** [reply]
感谢大佬，在webpack5 接 wasm 也遇到了问题，刚好看到umi这个解决方案，瞬间我的问题解决了
感谢

**Comment by deepdeepstudy** [comment]
your e-mail is recieved，have a nice day!

**Comment by deepdeepstudy** [comment]
your e-mail is recieved，have a nice day!
